src/views/darkroom.c

```diff
@@ -1881,6 +1890,12 @@ void leave(dt_view_t *self)
     free(module);
     dev->iop = g_list_delete_link(dev->iop, dev->iop);
   }
+  while(dev->alliop)
+  {
+    dt_iop_cleanup_module((dt_iop_module_t *)dev->alliop->data);
```

Comment:
can't we have multiple cleanup() for a module? this is done also just in the loop above.

Reply:
My bad, I didn't explain it right. dev->alliop don't work exactly like dev->allforms. For the masks, dev->allforms has ALL forms, and they are free() on exit, dev->alliop has only the modules that has been removed from dev->iop, and it assumes that dt_iop_cleanup_module() hasn't been called.
This is done here:
https://github.com/darktable-org/darktable/pull/1800/files#diff-2c39561107e06b0530706f865a77eabaR580

The idea is: the user delete a module, but the pipe may still need it, so, instead of physically delete it, it is removed from dev->iop (that's not an issue since the pipe works on a copy of dev->iop) and is added to dev->alliop, where it is free() on exit.